File: test/capture_test.cpp
#include <string>
#include "dogears/dogears.h"
#include "test_utilites.h"

int captureTest(dogears::DogEars& cape, unsigned int bufferSize, const std::string& name) {
    auto data = cape.capture<dogears::Normalized>(bufferSize);

    auto passed = true;
    passed &= assertEqual(data.channels(), 4u, name + ": Incorrect channel count");
    passed &= assertEqual(data.samples(), bufferSize, name + ": Incorrect sample count");

    passed &= checkBuffer(data, name);

    return passed;
}

int main(int argc, char* argv[]) {
    dogears::DogEars cape;

    auto passed = true;
    // Internalaly, the buffer size is some power of two
    passed &= captureTest(cape, 2048u, "Several complete buffers");
    passed &= captureTest(cape, 3000u, "Partial buffer");
    passed &= captureTest(cape, 10u, "Small capture size");

    printTestResults(passed);
}

File: test/capture_test.py
from dogears import *
from test_utilites import *

def captureTest(cape, bufferSize, errorMessagePrefix):
    data = cape.capture(bufferSize)

    passed = checkBuffer(data, errorMessagePrefix)
    passed &= assertEqual(data.shape[0], 4, "Incorrect number of channels")
    passed &= assertEqual(data.shape[0], bufferSize, "Incorrect number of samples")

with DogEars() as cape:
    passed = True

    # Check around the boundaries of the internal buffer, which has the
    # size of some multiple of two
    passed &= captureTest(cape, 2048, "Several complete buffers: ")
    passed &= captureTest(cape, 3000, "One incomplete buffer: ")

    passed &= captureTest(cape, 10, "One small buffer: ")

    printTestResults(passed)